packages/volto/src/middleware/api.js

```diff
@@ -188,7 +187,15 @@ const apiMiddlewareFactory =
                     ),
                   },
                 ).then((reqres) => {
-                  dispatch(updateUploadedFiles(uploadedFiles++));
+                  if (item.op === 'post') {
+                    let uploadedFiles =
+                      getState().content.subrequests?.['batch-upload']
+                        ?.uploadedFiles;
+
+                    dispatch(
+                      updateUploadedFiles(++uploadedFiles, 'batch-upload'),
```

Comment:
@victorchrollo14 the one getting the SUBREQUEST='batch-upload' is createContent action, when it's uploading files.

https://github.com/plone/volto/blob/8aa167a6d5ffb94f329dec42c0d9c96700ea27d6/packages/volto/src/components/manage/Contents/ContentsUploadModal.jsx#L187-L204

Why the `updateUploadedFiles` needs it to be passed down at all?

Reply:
> @victorchrollo14 the one getting the `SUBREQUEST='batch-upload'` is `createContent` action, when it's uploading files.
> 
> https://github.com/plone/volto/blob/8aa167a6d5ffb94f329dec42c0d9c96700ea27d6/packages/volto/src/components/manage/Contents/ContentsUploadModal.jsx#L187-L204
> 
> Why the `updateUploadedFiles` needs it to be passed down at all?

createContent does not handle updating the uploadedFiles count, So I created updateUploadedFiles action that accepts `subrequest, uploadedFiles` properties and a  reducer that takes care of updating the uploadedFiles count  which I have added into subrequests['batch-upload'].
What do you think? should we use `createContent` itself for updating  the uploadedFiles count.

Reply:
No no, the question is, why do you need to pass down the `subrequest` to `updateUploadedFiles`?